src/frontend/sharedComponents/LeaveProjectModalContent/LeaveProject.tsx

```diff
@@ -0,0 +1,193 @@
+import * as React from 'react';
+import {StyleSheet, View} from 'react-native';
+import ErrorIcon from '../../images/Error.svg';
+import {defineMessages, useIntl} from 'react-intl';
+import {Text} from '../../sharedComponents/Text';
+import {useLeaveProject} from '../../hooks/server/projects';
+import {RED} from '../../lib/styles';
+import MaterialIcons from 'react-native-vector-icons/MaterialIcons';
+import {TouchableOpacity} from '../../sharedComponents/Touchables';
+
+import {UIActivityIndicator} from 'react-native-indicators';
+import {BottomSheetModalContent} from '../BottomSheetModal';
+import {useAcceptInvite} from '../../hooks/server/invites';
+import {ErrorBottomSheet} from '../ErrorBottomSheet';
+
+const m = defineMessages({
+  leaveProj: {
+    id: 'screens.LeaveProject.leaveProj',
+    defaultMessage: 'Leave Project',
+  },
+  deleteConsentWithName: {
+    id: 'screens.LeaveProject.deleteConsentWithName',
+    defaultMessage:
+      'I understand I will be deleting all data from Project {projectName} from my device.',
+  },
+  deleteConsentWithoutName: {
+    id: 'screens.LeaveProject.deleteConsentWithoutName',
+    defaultMessage: 'I understand I will be deleting all data from my device.',
+  },
+  removeFromProjWithName: {
+    id: 'screens.LeaveProject.removeFromProjWithName',
+    defaultMessage:
+      "This will remove all Project {projectName}'s data from your device.",
+  },
+  removeFromProjWithoutName: {
+    id: 'screens.LeaveProject.removeFromProjWithoutName',
+    defaultMessage: 'This will remove all of the data from your device.',
+  },
+  cancel: {
+    id: 'screens.LeaveProject.cancel',
+    defaultMessage: 'Cancel',
+  },
+  checkToConfirm: {
+    id: 'screens.LeaveProject.checkToConfirm',
+    defaultMessage: 'Please check the box to confirm',
+  },
+  leavingProject: {
+    id: 'screens.LeaveProject.leavingProject',
+    defaultMessage: 'Leaving Project {projectName}',
+  },
+});
+
+type LeaveProjectProps = {
+  inviteId: string;
+  accept: ReturnType<typeof useAcceptInvite>;
+
+  closeSheet: () => void;
+  projectName?: string;
+};
+
+export const LeaveProject = ({
+  inviteId,
+  accept,
+  closeSheet,
+  projectName,
+}: LeaveProjectProps) => {
+  const {formatMessage} = useIntl();
+  const [error, setError] = React.useState(false);
+  const [isChecked, setIsChecked] = React.useState(false);
+  const leaveProject = useLeaveProject();
+  const [combinedLoading, setCombinedLoading] = React.useState(false);
```

Comment:
you may be able to use `leaveProject.isPending` instead of having a separate state.

Reply:
I noted it [here](https://github.com/digidem/comapeo-mobile/issues/525#issuecomment-2246123715) to fix when the leave project api is fixed.